### PR TITLE
on empty line-suffix, don't get completions

### PR DESCRIPTION
**Pull request by hlolli**
This is an old pain, but this is not a final solution as symbol names ending with alphanumeric number are legal symbols in Clojure as far as I can tell. Next step should be to modify `(re-find #":?([a-zA-Z-.<>*=&?]*|^\(/)$" line)` to something better (including number, or match all except xy). In any case tough, if this re-find doesn't find anything, (ie. invalid?) then completions should not continue, as empty string causes something like this

```
(time (do (lumo.repl/get-completions "red6" #(prn %)) nil))
d6qualified-ident?" "red6qualified-keyword?" "red6qualified-symbol?" "red6quot" "red6rand" "red6rand-int" "red6rand-nth" "red6random-sample" "red6random-uuid" "red6range" "red6ranged-iterator" "red6re-find" "red6re-matches" "red6re-pattern" "red6re-seq" "red6realized?" "red6record?" "red6reduce" "red6reduce-kv" "red6reduceable?" "red6reduced" "red6reduced?" "red6reductions" "red6refer-clojure" "red6regexp?" "red6reify" "red6rem" "red6remove" "red6remove-all-methods" "red6remove-method" "red6remove-watch" "red6repeat" "red6repeatedly" "red6replace" "red6replicate" "red6require" "red6require-macros" "red6reset!" "red6reset-meta!" "red6reset-vals!" "red6resolve" "red6rest" "red6reverse" "red6reversible?" "red6rseq" "red6rsubseq" "red6run!" "red6satisfies?" "red6second" "red6select-keys" "red6seq" "red6seq-iter" "red6seq?" "red6seqable?" "red6sequence" "red6sequential?" "red6set" "red6set-from-indexed-seq" "red6set-print-err-fn!" "red6set-print-fn!" "red6set-validator!" "red6set?" "red6short" "red6shorts" "red6shuffle" "red6simple-benchmark" "red6simple-ident?" "red6simple-keyword?" "red6simple-symbol?" "red6some" "red6some->" "red6some->>" "red6some-fn" "red6some?" "red6someprotopackage.TestPackageTypes" "red6sort" "red6sort-by" "red6sorted-map" "red6sorted-map-by" "red6sorted-set" "red6sorted-set-by" "red6sorted?" "red6source" "red6special-symbol?" "red6specify" "red6specify!" "red6split-at" "red6split-with" "red6spread" "red6str" "red6string-hash-cache" 
..elided...
"Elapsed time: 553.308537 msecs"
```

### CODE REVIEW

**Comment by anmonteiro** [comment]
clearly tests are failing

**Comment by hlolli** [comment]
`node target/bundle.js -c target/ -n ""`

```
Lumo 1.8.0-beta
ClojureScript 1.9.946
Node.js v9.2.0
 Docs: (doc function-name-here)
       (find-doc "part-of-name-here")
 Source: (source function-name-here)
 Exit: Control+D or :cljs/quit or exit

Error: Port should be > 0 and < 65536. Received NaN.
cljs.user=> 
```
Same output for latest release and the dev build, the test is expecting
```
"Error: "port" argument must be >= 0 and < 65536
```
I'm clueless how this and my last commit could have caused this change. Different node version?

**Comment by hlolli** [comment]
Should I edit the snapshot, or modify the error message?

**Comment by arichiardi** [comment]
Was it the node version the problem?

**Comment by arichiardi** [comment]
Node version might have changed the message, see #334

**Comment by arichiardi** [comment]
@hlolli I fixed the problems with CI, I was wondering if you can rebase and retry

File: src/cljs/snapshot/lumo/repl.cljs [hunk 1]
```diff
@@ -1445,12 +1445,14 @@
           ns-alias (second (re-find #"\(*(\b[a-zA-Z-.<>*=&?]+)/[a-zA-Z-]*$" line))
           line-match-suffix (first (re-find #":?([a-zA-Z-.<>*=&?]*|^\(/)$" line))
           line-prefix (subs line 0 (- (count line) (count line-match-suffix)))
-          completions (reduce (fn [ret item]
-                                (doto ret
-                                  (.push (str line-prefix item))))
+          completions (if (empty? line-match-suffix)
```
Comment (arichiardi) [inline src/cljs/snapshot/lumo/repl.cljs]: I am surprised that the `completion-candidates` fn does not depend on the prefix item. I would expect it is it's responsibility to do the `if`. Let me try to push a commit here.

Comment (arichiardi) [reply src/cljs/snapshot/lumo/repl.cljs]: Oh I see `completion-candidates` comes up with all the completions all the time

**Comment by arichiardi** [comment]
I took a little bit of time to check what is going on here and opened an alternative branch. Hopefully the fixes are good enough. We cannot filter out everything when `line-match-suffix` is empty because we would also kill legit completions for namespaces/built ins/`goog`.

**Comment by anmonteiro** [comment]
I'm going to close this in favor of #337

**Pull request closed without merging**